File: check_keys.js
const fs = require('fs');
const content = fs.readFileSync('app.js', 'utf8');

// Use regex to roughly extract bibleData and tamilBibleBookNames
const bibleDataMatch = content.match(/const bibleData = \{([\s\S]*?)\n\};\n/);
const tamilNamesMatch = content.match(/const tamilBibleBookNames = \{([\s\S]*?)\n\};\n/);

if (!bibleDataMatch || !tamilNamesMatch) {
  console.log('Failed to match');
  process.exit();
}

const lines1 = bibleDataMatch[1].split('\n').map(l => l.trim()).filter(l => l.startsWith('\"'));
const lines2 = tamilNamesMatch[1].split('\n').map(l => l.trim()).filter(l => l.startsWith('\"'));

const keys1 = lines1.map(l => l.match(/\"([^\"]+)\"/)[1]);
const keys2 = lines2.map(l => l.match(/\"([^\"]+)\"/)[1]);

const onlyIn1 = keys1.filter(k => !keys2.includes(k));
const onlyIn2 = keys2.filter(k => !keys1.includes(k));

console.log('In bibleData but not in tamilBibleBookNames:', onlyIn1);
console.log('In tamilBibleBookNames but not in bibleData:', onlyIn2);

File: check_keys.py
import re

with open('app.js', 'r', encoding='utf-8') as f:
    text = f.read()

m1 = re.search(r'const bibleData = \{(.*?)\n\};', text, re.DOTALL)
m2 = re.search(r'const tamilBibleBookNames = \{(.*?)\n\};', text, re.DOTALL)

lines1 = [x.strip() for x in m1.group(1).split('\n') if x.strip().startswith('"')]
lines2 = [x.strip() for x in m2.group(1).split('\n') if x.strip().startswith('"')]

keys1 = [re.search(r'"([^"]+)"', x).group(1) for x in lines1]
keys2 = [re.search(r'"([^"]+)"', x).group(1) for x in lines2]

print("In bibleData not in dict:", [k for k in keys1 if k not in keys2])
print("In dict not in bibleData:", [k for k in keys2 if k not in keys1])
